jfrog-cli/artifactory/commands/golang/go.go

```diff
@@ -114,12 +124,51 @@ func ExecuteGo(noRegistry bool, goArg, targetRepo, buildName, buildNumber string
 		if err != nil {
 			return err
 		}
+		err = goProject.LoadDependencies()
+		if err != nil {
+			return err
+		}
 		err = utils.SaveBuildInfo(buildName, buildNumber, goProject.BuildInfo(false))
 	}
 
 	return err
 }
 
+func retrieveAndPublish(targetRepo, goArg string, depsTidy bool, details *config.ArtifactoryDetails) error {
+	// Need to run Go without Artifactory to resolve all dependencies.
+	cache := goutils.GetStaticCache()
+	wd, rootDir, err := collectDependenciesPopulateAndPublish(targetRepo, depsTidy, &cache, details)
+	if err != nil {
+		if depsTidy {
+			log.Debug("Received and error:", err)
+		} else {
+			return err
+		}
+	}
+	// Lets run the same command again now that all the dependencies were downloaded.
+	// Need to run only if the command is not go mod download or go mod tidy since this was run by the CLI to download and publish to Artifactory
+	log.Info("Finished with following stats:", cache.GetSuccess(), "/", cache.GetTotal())
```

Comment:
I suggest rephrasing the log message to:
// Done building and publishing X go dependencies to Artifactory out of a total of Y dependencies.

Reply:
Done